app/src/main/java/com/appunite/loudius/network/utils/AuthFailureHandler.kt

```diff
@@ -0,0 +1,23 @@
+package com.appunite.loudius.network.utils
+
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import javax.inject.Inject
+import javax.inject.Singleton
+
+interface AuthFailureHandler {
+    suspend fun emitAuthFailure()
+
+    val authFailureFlow: SharedFlow<Unit>
+}
+
+@Singleton
+class AuthFailureHandlerImpl @Inject constructor() : AuthFailureHandler {
```

Comment:
suggestion: you provide this class through Hilt's `@Provide` so it's redundant
```suggestion
class AuthFailureHandlerImpl : AuthFailureHandler {
```